### ISSUE TITLE
Default embedded MP4 subtitles not always displayed by default

### ISSUE THREAD

**Issue by JNavas2** [clsid2/mpc-hc#2210]
I want subtitles to always display by default.
MP4 video files with embedded subtitles:
- Display by default on files created with VidCoder (Handbrake engine).
- Do NOT display by default on files created with ffmpeg.

and I can't figure out the difference. Default flag in ON in both.
"Prefer forced and/or default subtitles tracks" IS checked.
Any thoughts?
Forced option is weird in VidCoder file, but that shouldn't matter, right?

**MPC-HC DISPLAYS AUTOMATICALLY (VIDCODER)**

<img width="342" alt="notepad_XX3NNS3PSn" src="https://github.com/clsid2/mpc-hc/assets/14850792/7bf0b4ee-3246-4828-b707-0a541baac0a7">

```
Text
ID                                       : 3
Format                                   : Timed Text
Muxing mode                              : sbtl
Codec ID                                 : tx3g
Duration                                 : 51 min 26 s
Bit rate mode                            : Variable
Bit rate                                 : 110 b/s
Frame rate                               : 0.343 FPS
Stream size                              : 41.3 KiB (0%)
Default                                  : Yes
Forced                                   : Yes / No
Alternate group                          : 3
Encoded date                             : 2023-08-04 23:32:48 UTC
Tagged date                              : 2023-08-04 23:32:48 UTC
Count of events                          : 751
```

**MPC-HC DOES NOT DISPLAY AUTOMATICALLY (FFMPEG)**

<img width="333" alt="notepad_ATZ5xfI6zN" src="https://github.com/clsid2/mpc-hc/assets/14850792/1dcd6bd2-3d4f-4a72-a878-301bf8274281">

```
Text
ID                                       : 3
Format                                   : Timed Text
Muxing mode                              : sbtl
Codec ID                                 : tx3g
Duration                                 : 51 min 8 s
Bit rate mode                            : Variable
Bit rate                                 : 104 b/s
Frame rate                               : 0.400 FPS
Stream size                              : 38.8 KiB (0%)
Language                                 : English
Default                                  : Yes
Forced                                   : No
Alternate group                          : 3
Count of events                          : 1021
```

**Comment by clsid2** [comment]
Options > Internal Filters > Splitter
There you can configure which embedded subtitle track should be selected by default.

If default flag in your MP4 files is not recognized, you need to submit bug report to FFmpeg (which is used by LAV Filters). I can't help you with that as it is not a player issue.

**Comment by JNavas2** [comment]
With respect, there is no apparent ffmpeg issue.
MediaInfo output for the ffmpeg file shows the Default flag to be set.
But I've now managed to solve my problem, by adding "eng" to "Languages in order of preference".
That wasn't needed for VidCoder files, but it is needed for ffmpeg files.
My guess is that subtitle flags are more accurately set by ffmpeg than by VidCoder (Handbrake).

**Comment by JNavas2** [comment]
This issue also appears to be FIXED in 2.2.1.72
👍👏

**Comment by tjoynt** [comment]
Thanks @JNavas2, your comment helped me solve my own problem! I had a similar issue, where the subtitles did not display by default with .mp4 files that I created. I fixed it by specifying a language when I muxed the video+audio and subtitle (.srt) files together using ffmpeg: I added "-metadata:s:s:0 language=eng" before the output file name. 
Full command: ffmpeg -i videoFile -i subtitleFile -c copy -c:s mov_text -metadata:s:s:0 language=eng  outputFile
